operator/helper/ip_resolver.go

```diff
@@ -0,0 +1,31 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helper
+
+import "net"
+
+func LookupIpAddr(ip string) (host string) {
+	res, err := net.LookupAddr(ip)
```

Comment:
Yeah, I made probably invalid assumption that this is cached by system. Do you have some recommendation regarding cache implementation/cache libraries, or should I just use map pointing to object with name and timestamp?

Reply:
I'm not familiar with the implementation, so perhaps caching is built in.

Assuming not though, I believe a map as described would be idiomatic.

Reply:
I pushed cached version. I have some concerns about that:
 - is making IPResolver global a good approach? Not really sure how to make all of: common cache, invalidations and stop it nicely
 - is `helper.IPResolver.GetHostFromIp(ip)` acceptable way of using helpers in this repo?

Reply:
What would the memory footprint of this cache be, at say, 100k TCP connections? Should there be an upper limit to the cache's size?

Im curious because this week I did some connections testing with Stanza's TCP operator, which will also benefit from this.

Reply:
I might be thinking too much on this, but should there be a cache interface that allows us to extend this implementation to something like Redis? 

I think in memory is likely fine for most usecases, but having the option to implement something externally later would be nice.

I just wanted to throw the idea out while there is a discussion around it. This is probably a bigger effort outside of the scope of this PR.

Reply:
I would avoid the global. You already have `NewIpResolver()`, so might as well attach to the UPD/TCP operator struct. Only needs to be instantiated if `add_attributes=true` though. This change should also make accessing the IPResolver more natural.

As far as how to invalidate the cache. It seems an initial implementation can just be as simple as locking the mutex, instantiating a new map, unlock mutex. I think the rest looks good.

Reply:
@jsirianni, do you think 100k unique ip's is possible, or would many be duplicate (and therefore be cache hits)?

Reply:
@sumo-drosiek I would avoid the global, and just attach a resolver to each operator. This should make calling it more natural too.

I think your cache purge is fine for now. Maybe we'll have a more nuanced invalidation logic later, but this seems reasonable for a first pass.

Reply:
@djaglowski I think it is possible but unlikely. I only mentioned it because I was able to do it without issue.

Reply:
I think using other cache is a good idea, but as mentioned it's not a purpose of this PR. Basing on some smoke tests, I think that `100 000` should take a few MB of memory for IPv4

Reply:
some smoke tests:
```
package helper

import (
	"fmt"
	"testing"
	"unsafe"
)

func TestIPResolverLookup(t *testing.T) {
	resolver := NewIpResolver()
	for i := 0; i < 256; i++ {
		ip := resolver.lookupIpAddr(fmt.Sprintf("192.168.1.%d", i))
		fmt.Printf("%v\n", ip)
	}

	size := unsafe.Sizeof(resolver.cache)
	for i := 0; i < 256; i++ {
		ip := resolver.cache[fmt.Sprintf("192.168.1.%d", i)]
		size += unsafe.Sizeof(ip)
		size += unsafe.Sizeof(ip.hostname)
		size += unsafe.Sizeof(ip.expireTime)
	}
	fmt.Printf("all cache size: %v", size)
}
```

```
...
ip-192-168-1-248.us-west-1.compute.internal
ip-192-168-1-249.us-west-1.compute.internal
ip-192-168-1-250.us-west-1.compute.internal
ip-192-168-1-251.us-west-1.compute.internal
ip-192-168-1-252.us-west-1.compute.internal
ip-192-168-1-253.us-west-1.compute.internal
ip-192-168-1-254.us-west-1.compute.internal
ip-192-168-1-255.us-west-1.compute.internal
all cache size: 20488
```

Reply:
I revisited my calculation based on this: https://man7.org/linux/man-pages/man7/hostname.7.html

|  variable | size (B) |
|---|---|
| cacheEntry struct size | 40 |
| max string length | 256 |
| max total size | 296 |
| total size of 100 000 entries | 28.228759765625 MB |

Reply:
I don't have data to support this, but my intuition is that 99.9% of cases will involve receiving data from less than 100 unique IPs, in which case the cache is less than 3kB. Does that seem like a reasonable assumption for now?

If someone has a 100_000 ip use case, then I think that warrants a separate PR.